dbms/src/Storages/transformQueryForExternalDatabase.cpp

```diff
@@ -20,31 +23,64 @@ namespace ErrorCodes
     extern const int LOGICAL_ERROR;
 }
 
-static void replaceConstFunction(IAST & node, const Context & context, const NamesAndTypesList & all_columns)
+namespace
 {
-    for (size_t i = 0; i < node.children.size(); ++i)
+
+class ReplacingConstantExpressionsMatcher
+{
+public:
+    using Data = Block;
```

Comment:
Interesting hack. I've never thought about assigning Data to a real class.

Reply:
I didn't consider it as hack. I have just read ActionsVisitor and have thought that it's the intended way of usage.